### PR TITLE
Table component

### PR DESCRIPTION
**Pull request by adamzerella**
The first release of the `table` component takes inspiration from a lot of the community discussion, other notable design systems and previous work last year. We decided for the first release we would ship a vertical table layout e.g:

| heading 1 | heading 2 |
| --- | --- |
| row 1, col 1, cell 1 | row 1, col 2, cell 2 |
| row 2, col 1, cell 3 | row 2, col 2, cell 4 |
| row 3, col 1, cell 5 | row 3, col 2, cell 6 |

Community discussion: https://community.digital.gov.au/t/table/92/20

## Community credits / thanks:
@dkeeghan
@TrebBrennan for https://codepen.io/AusDesignSystem/pen/PeGwjr
@steveworley + @tobybellwood for #202 
@alex-page
@AndrewArch
@Chris820
@rooby

## Related issues: 
#393 
#202 
#8
#552

### CODE REVIEW

File: packages/table/tests/site/test.scss [hunk 1]
```diff
@@ -4,7 +4,7 @@
 //testing includes
 @import '../../../core/src/sass/_module.scss';
 
-@import '../../../grid/src/sass/_module.scss';
+@import '../../../grid-12/src/sass/_module.scss';
```
Comment (adamzerella) [inline packages/table/tests/site/test.scss]: @sukhrajghuman We should fix this in template

Comment (adamzerella) [reply packages/table/tests/site/test.scss]: nvm

File: packages/table/tests/site/test.scss [hunk 2]
```diff
@@ -6,5 +6,13 @@
...
 @import '../../../grid-12/src/sass/_module.scss';
 
+@import '../../../body/src/sass/_module.scss';
+
+
 //styling all test files
 @import '../../../../.templates/test-assets/_style.scss';
+
+
+.table-wrapper {
```
Comment (adamzerella) [inline packages/table/tests/site/test.scss]: Shouldn't this just be in the `table` selector ?

**Comment by sukhrajghuman** [comment]
thanks @maxious have made some updates accordingly. Now the order of the row data can be in any order and have removed the `headerContentTypes` prop

**Pull request merged**